src/App.tsx

```
@@ -8,53 +8,185 @@ import { PostsList } from './components/PostsList';
-                className="notification is-danger"
-                data-cy="PostsLoadingError"
-              >
-                Something went wrong!
-              </div>
+  const [error, setError] = useState(false);
+  const [errorComments, setErrorComments] = useState(false);
+
+  const [isUserSelected, setIsUserSelected] = useState(false);
+  const [selectedUserId, setSelectedUserId] = useState<number | null>(null);
+
+  const [isSidebarOpen, setIsSidebarOpen] = useState(false);
+
+  const [comments, setComments] = useState<Comment[]>([]);
+
+  async function fetchCreateComment({
+    name,
+    email,
+    body,
+    postId,
+  }: Omit<Comment, 'id'>) {
+    try {
+      const result = await createComments({ name, email, body, postId });
+
+      setComments(currentComments => [...currentComments, result]);
+    } catch (errorCreatComments) {
+      throw errorCreatComments;
+    }
+  }
+
+  async function fetchPosts(userId: number) {
+    setIsLoading(true);
+    setError(false);
+    setIsUserSelected(!!userId);
+
+    if (userId !== selectedUserId) {
+      setIsSidebarOpen(false);
+      setSelectedUserId(userId);
+    }
+
+    try {
+      const results = await getPosts(userId);
+
+      setPosts(results);
+    } catch {
+      setError(true);
+    } finally {
+      setIsLoading(false);
+    }
+  }
+
+  async function fetchComments(postId: number) {
+    setIsLoadingComments(true);
+    setErrorComments(false);
+
+    try {
+      const results = await getComments(postId);
+
+      setComments(results);
+    } catch {
+      setErrorComments(true);
+    } finally {
+      setIsLoadingComments(false);
+    }
+  }
 
-              <div className="notification is-warning" data-cy="NoPostsYet">
-                No posts yet
+  async function fetchDeleteComment(id: number) {
+    setComments(currentComments =>
+      currentComments.filter(comment => comment.id !== id),
+    );
+
+    try {
+      await deleteComments(id);
+    } catch (errDelete) {
```

This catch rethrows the caught error (`throw errDelete;`) without logging or updating any UI-visible error state. As above: "Replace empty catch blocks with at least `console.error(error)` and, where appropriate, set error state." If delete failures should be surfaced to the user (or retried), set an error state and/or log the error. If you intentionally want to propagate the error, ensure the call site will catch it and handle it; otherwise logging plus a user-visible error is preferred.